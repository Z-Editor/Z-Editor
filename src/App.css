.public-DraftEditor-content > div > *:first-child{
  margin-top: 1px;
}
.public-DraftEditor-content > div > *:not(:first-child){
  margin-top: 0px;
}

.header {
  font-family: "Lucida Grande";
  font-weight: 700;
  line-height: 1.0;
  margin-top: 39px;
  margin-bottom: 0;
}
.h1{
  font-size: 52px;
}
.h2{
  font-size: 44px;
}
.h3{
  font-size: 36px;
}
.h4{
  font-size: 28px;
}
.header + .header, .header + * + .header {
  margin-top: 1px!important;
  font-weight: initial;
  color: rgba(0,0,0,.44);
}
.header + .paragraph, .header + * + .paragraph {
  margin-top: 10px!important;
}

.Zeditor-container {
  width: 100%;
  height: 100%;
}

.Zeditor-root {
  background: white;
  -webkit-font-smoothing: antialiased;
  margin: 40px auto;
  width: 700px;
}

.Zeditor-editor {
  cursor: text;
  font-size: 15px;
  line-height: 26px;
  min-height: 34px;
  padding: 25px;
}

.Zeditor-button {
  margin-top: 10px;
  text-align: center;
}

.Zeditor-handle {
  color: rgba(98, 177, 254, 1.0);
  direction: ltr;
  unicode-bidi: bidi-override;
}

.Zeditor-hashtag {
  color: rgba(95, 184, 138, 1.0);
}

.Zeditor-tex {
  background-color: #fff;
  cursor: pointer;
  margin: 20px auto;
  padding: 20px;
  -webkit-transition: background-color 0.2s fade-in-out;
  user-select: none;
  -webkit-user-select: none;
}

.Zeditor-activeTeX {
  color: #888;
}

.Zeditor-panel {
  font-family: 'Helvetica', sans-serif;
  font-weight: 200;
}

.Zeditor-panel .Zeditor-texValue {
  border: 1px solid #e1e1e1;
  display: block;
  font-family: 'Inconsolata', 'Menlo', monospace;
  font-size: 14px;
  height: 110px;
  margin: 20px auto 10px;
  outline: none;
  padding: 14px;
  resize: none;
  -webkit-box-sizing: border-box;
  width: 500px;
}

.Zeditor-buttons {
  text-align: center;
}

.Zeditor-saveButton,
.Zeditor-removeButton {
  background-color: #fff;
  border: 1px solid #0a0;
  cursor: pointer;
  font-family: 'Helvetica', 'Arial', sans-serif;
  font-size: 16px;
  font-weight: 200;
  margin: 10px auto;
  padding: 6px;
  -webkit-border-radius: 3px;
  width: 100px;
}

.Zeditor-removeButton {
  border-color: #aaa;
  color: #999;
  margin-left: 8px;
}

.Zeditor-invalidButton {
  background-color: #eee;
  border-color: #a00;
  color: #666;
}

.Zeditor-insert {
  background-color: #f1f1f1;
  border: 1px solid #ccc;
  border-radius: 3px;
  bottom: 30px;
  position: fixed;
  right: 30px;
}
.Zeditor-insert2 {
  background-color: #f1f1f1;
  border: 1px solid #ccc;
  border-radius: 3px;
  bottom: 30px;
  position: fixed;
  right: 160px;
}

.Zeditor-insert3 {
  background-color: #f1f1f1;
  border: 1px solid #ccc;
  border-radius: 3px;
  bottom: 30px;
  position: fixed;
  right: 300px;
}

.logo {
  float: left;
  font-size: 30px;
  color: white;
  margin: 15px;
}

.github-button {
  float: left;
  margin: 15px;
  margin-top: 26px;
  color: white;
}

.flex-container {
  margin-top: 100px;
}

.head {
  /*background: linear-gradient(150deg, #ffb347, #ffcc33);*/
  background: linear-gradient(150deg,#3366ff, #3333cc);

  height: 65px;
  top:0;
  width: 100%;
  z-index: 10;
  position: fixed;
}

.container-content {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.button {
  outline: 0;
  cursor: pointer;
  float: right;
  margin: 12px;
  background: transparent;
  border-radius: 30px;
  border: 1px solid transparent;
  /* font-weight: bold; */
  font-size: 15px;
  color: rgb(255, 255, 255);
  padding: 10px;
}
.button:hover{
  border: 1px solid rgba(255, 255, 255, 0.51);
}
.button.active{
  border: 1px solid rgba(255, 255, 255, 0.51);
}

.block {
  margin-bottom: 10px;
}

.schemata {
  border-left: 2px solid black;
  position: relative;
  padding: 1px 20px;
  /*box-shadow: inset 0px 0px 10px 2px #f7f7f7;*/
}

.schemata_up {
  border-bottom: 2px solid black;
  position: relative;
  padding: 8px 20px;
  padding-left:10px;
  line-height:0.001em;
  
}

.schemata_down {
  border-left: 2px solid black;
  border-bottom: 2px solid black;
  position: relative;
  padding: 1px 20px;
  /*box-shadow: inset 0px 0px 10px 2px #f7f7f7;*/
}

.schemata_up span {
  padding:10px 10px;
  background:#fff;

}

body {
  font-family: "Georgia","Cambria","Times New Roman","Times","serif";
  line-height: initial;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  background-color: #FAFAFA;
}
* {
  box-sizing: border-box;
  -moz-box-sizing: border-box;
}
.page {
  width: 210mm;
  min-height: 297mm;
  padding: 20mm;
  margin: 10mm auto;
  margin-top:150px;
  border: 1px #D3D3D3 solid;
  border-radius: 5px;
  background: white;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
}
.subpage {
  padding: 1cm;
  border: 5px red solid;
  height: 257mm;
  outline: 2cm #FFEAEA solid;
}

@page {
  size: A4;
  margin: 0;
}

@media print {  
  .head {
    display: none;
  }
  
  .menu * {
    display: none;
  }
  .menu_left * {
    display: none;
  }
  body * {
    visibility: hidden;
  }
  .rdw-editor-toolbar{
    display: none;
  }
  .page * {
    visibility: visible;
  }
  .page {
    background-color: white;
    height: 100%;
    width: 100%;
    position: relative;
    top: -60px;
    left: 0;
    margin: 0;
    padding: 20px 45px 20px 45px;
    font-size: 14px;
    line-height: 18px;
  }
}

.menu {
  position: fixed;
  right: 0;
  top: 30%;
  width: 17em;
  margin-top: -2.5em;
}

.menu_left {
  position: fixed;
  left: -40px;
  top: 30%;
  width: 13em;
  margin-top: -2.5em;
}
