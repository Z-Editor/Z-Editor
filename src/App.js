import React, { Component } from 'react';
import ZEditor from './editor/editor';
import TopBar from './TopBar';
import './App.css';

class App extends Component {

    render() {
        return (
            <div className="flex-container">

                <TopBar />

                <ZEditor />

            </div>
        );
    }
}

export default App;
