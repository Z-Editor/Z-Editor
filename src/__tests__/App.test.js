import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';

configure({ adapter: new Adapter() });

describe('renders without crashing', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  describe('button functionality', () => {
    it('exists', () => {
      expect(app.find('button')).toBeTruthy;
    });
    it('calls window.print on click', () => {
      global.print = jest.fn();
      const button = app.find('button');
      button.simulate('click');
      expect(window.print).toHaveBeenCalled();
    });
  });
});
