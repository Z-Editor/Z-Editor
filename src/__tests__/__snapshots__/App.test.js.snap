// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="head"
        >
          <div
            className="logo"
          >
            Z-Editor
          </div>
          <a
            className="github-button"
            href="https://github.com/Z-Editor/Z-Editor"
            rel="noopener noreferrer"
            target="_blank"
          >
            View on Github
          </a>
          <button
            className="button"
            onClick={[Function]}
          >
            Print
          </button>
        </div>,
        <ZEditor />,
      ],
      "className": "flex-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="logo"
            >
              Z-Editor
            </div>,
            <a
              className="github-button"
              href="https://github.com/Z-Editor/Z-Editor"
              rel="noopener noreferrer"
              target="_blank"
            >
              View on Github
            </a>,
            <button
              className="button"
              onClick={[Function]}
            >
              Print
            </button>,
          ],
          "className": "head",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Z-Editor",
              "className": "logo",
            },
            "ref": null,
            "rendered": "Z-Editor",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View on Github",
              "className": "github-button",
              "href": "https://github.com/Z-Editor/Z-Editor",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "View on Github",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Print",
              "className": "button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Print",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="head"
          >
            <div
              className="logo"
            >
              Z-Editor
            </div>
            <a
              className="github-button"
              href="https://github.com/Z-Editor/Z-Editor"
              rel="noopener noreferrer"
              target="_blank"
            >
              View on Github
            </a>
            <button
              className="button"
              onClick={[Function]}
            >
              Print
            </button>
          </div>,
          <ZEditor />,
        ],
        "className": "flex-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="logo"
              >
                Z-Editor
              </div>,
              <a
                className="github-button"
                href="https://github.com/Z-Editor/Z-Editor"
                rel="noopener noreferrer"
                target="_blank"
              >
                View on Github
              </a>,
              <button
                className="button"
                onClick={[Function]}
              >
                Print
              </button>,
            ],
            "className": "head",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Z-Editor",
                "className": "logo",
              },
              "ref": null,
              "rendered": "Z-Editor",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "View on Github",
                "className": "github-button",
                "href": "https://github.com/Z-Editor/Z-Editor",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "View on Github",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Print",
                "className": "button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Print",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
